presto-spi/src/main/java/com/facebook/presto/spi/type/UnscaledDecimal128Arithmetic.java

```diff
@@ -500,6 +557,11 @@ private static void setNegative(Slice decimal, int v3, boolean negative)
         setRawInt(decimal, SIGN_INT_INDEX, v3 | (negative ? SIGN_INT_MASK : 0));
     }
 
+    private static void setNegative(Slice decimal, long hi, boolean negative)
```

Comment:
Having two methods:

`setNegative(Slice, long, boolean)` and `setNegative(Slice, int, boolean)` is very error prone. Use different names for both.